app/scripts/modules/kubernetes/loadBalancer/configure/wizard/advancedSettings.html

```diff
@@ -76,4 +76,8 @@
     <map-editor model="loadBalancer.serviceAnnotations" label="Service Annotations" add-button-label="Add Annotations" labels-left="true">
     </map-editor>
   </div>
+  <div class="col-md-12">
+    <map-editor model="loadBalancer.serviceLabels" label="Service Labels" add-button-label="Add Label" labels-left="true">
```

Comment:
Related to your corresponding Clouddriver changes - if I clear all the service labels here while editing a load balancer, what does Clouddriver do? (The way it's implemented now, I think you'll inherit the existing labels.)

Reply:
That's correct. Hmmm, guess I should implement it so that the labels get cleared.

Reply:
I'm not sure, I think it takes the empty set. In any case, the logic is the same as it currently is for annotations.

Reply:
From Clouddriver:

```groovy
    def labels = description.serviceLabels ?: existingService?.metadata?.labels

    serviceBuilder = serviceBuilder.withLabels(labels)

    def annotations = description.serviceAnnotations ?: existingService?.metadata?.annotations

    serviceBuilder = serviceBuilder.withAnnotations(annotations)
```

Reply:
Yeah, I think passing an empty object from Deck is the right behavior. But the Clouddriver implementation is wrong - should be ternary on `description.serviceLabels == null`.

If you could also fix this for annotations (in Clouddriver), that would be great.

Reply:
You mean like this?

```groovy
 def annotations = description.serviceAnnotations == null ? existingService?.metadata?.annotations : description.serviceAnnotations
```

Reply:
Yes, exactly. Thank you.

Reply:
Done: https://github.com/spinnaker/clouddriver/pull/2014/commits/37a9dfbbc481b1460484a4bc3e7fcf616b4d0361#diff-6d44078ab20702f2e88f8d27785bfc33